utilcode/src/main/ets/utils/LocationUtils.ets

```diff
@@ -0,0 +1,282 @@
+/*
+* Copyright (C) 2022 Application library engineering group.
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import systemGeolocation from '@system.geolocation';
+import geolocation from '@ohos.geolocation';
+import ability_featureAbility from '@ohos.ability.featureAbility'
+
+export class LocationUtils {
+  mYLocationListener: myLocationListener = null ;
+  mListener: OnLocationChangeListener = null;
+  constructor() {
+    console.info(` u can't instantiate me...`)
+  }
+
+  isGpsEnabled() {
+    var context = ability_featureAbility.getContext();
+    let permission = "ohos.permission.LOCATION";
+    let options = {
+      "pid": 12,
+      "uid": 0,
+    };
+    context.verifyPermission(permission, options)
+      .then((data) => {
+        console.info('permission is granted ');
+      }).catch((error) => {
+      console.error('permission is not granted ');
+    })
+    if (geolocation.isGeoServiceAvailable()) {
+      console.log('GPS is enabled !')
+      return true
+    } else {
+      console.log('Gps is Not available ')
+      return false
+    }
+  }
+
+  isLocationEnabled() {
+    var context = ability_featureAbility.getContext();
+    let permission = "ohos.permission.LOCATION";
```

Comment:
permission check code move to a common function and call that function, otherwise its lot of code duplication.
Also if the permission is not granted need to ask user to grant permission by showing popup and if he denies we should not proceed further and return.